src/components/AppBar/AppBar.jsx

```diff
@@ -61,14 +72,20 @@ const AppBar = ({
 }) => {
   const [showDrawer, setShowDrawer] = useState(false);
   const nameContext = useContext(NameContext);
+  const userContext = useContext(UserContext);
   const history = useHistory();
 
   const pageTitle = (routes.find((item) => item.key === selected) || {}).label;
 
   const goBack = () => {
     history.goBack();
   };
-
+  // if there is no username, the user might not be signed in
+  // so fallback to icon
+  let avatar = <PersonIcon color="primary" />;
+  if (userContext.user.username) {
+    avatar = userContext.user.username[0];
```

Comment:
Is it the name or username that populates the avatar?

Reply:
We will keep username for now